spring-core/src/main/java/org/springframework/core/codec/CharSequenceEncoder.java

```diff
@@ -75,9 +74,21 @@ public boolean canEncode(ResolvableType elementType, @Nullable MimeType mimeType
 					return Hints.getLogPrefix(hints) + "Writing " + formatted;
 				});
 			}
-			CharBuffer charBuffer = CharBuffer.wrap(charSequence);
-			ByteBuffer byteBuffer = charset.encode(charBuffer);
-			return bufferFactory.wrap(byteBuffer);
+			boolean release = true;
+			DataBuffer dataBuffer = bufferFactory.allocateBuffer();
```

Comment:
We're allocating here a buffer with the default size. The problem is, there's no consensus about what capacity should be allocated. Some implementations favor "string length x average byte count per char with that charset", others (like Netty) directly go for "string length x max byte count per char with that charset".